maturity-model.md

```diff
@@ -0,0 +1,175 @@
+## Title
+Maturity model: Learn about InnerSource best practices
+
+## Patlet
+
+Teams have started adopting InnerSource. The practice is spreading to multiple
+departments. Understanding of what constitutes an InnerSource project are wide
+spread though. The solution is to provide a maturity model to allow for teams to
+go through a self check and discover patterns and practices that they are not
+yet aware of.
+
+## Problem
+
+When InnerSource adoption in an enterprise starts to increase individual
+mentoring of each project through one evangelist becomes unfeasible. Also more
+and more people are gaining at least a basic understanding of what it means to
+work in an InnerSource project. Looking at all InnerSource projects though the
+depth of understanding for the concept will diverge. Teams may not be aware of
+proven patterns that would help them move to the next level and solve issues and
+pain points that they are dealing with.
+
+## Context
+
+Several teams have started adopting InnerSource practices. The exact level of
+understanding of the practice diverges between teams. The exact problems teams
+run into diverge depending on the context and working environment of each team.
+As a result the definition of what are important best practices in an
+InnerSource project differs depending on each team.
+
+## Forces
+
+Teams sharing InnerSource learnings run into misunderstandings as they are not
+aware of their respective level of InnerSource adoption.
+
+Teams believe that "it's all about migrating to a shared software development
+forge".
+
+Teams are not aware of best practices that would help them solve issues that
+they run into in their daily doing.
+
+## Solution
+
+Ask teams to self assess against the proposed maturity model.
+
+### Transparency
+
+**Plans & Products**
+
+* PP-0: Individuals and teams do not regularly disclose their plans or products to multiple stakeholders. No formal actions exists at the organization.
+* PP-1: Individuals and teams give visibility to their plans or products to multiple stakeholders, before they are started. Shared roadmap.
+* PP-2: Stakeholders can contribute to some plans or products.
+* PP-3: There is an standard process for collective product and plans definition.
+
+
+**Development Process & Tools**
+
```

Comment:
Regarding to your second comment, I'd say that it is good that each InnerSource project defines its own process, but at the same time, one of the goals of InnerSource is to create homogeneous ways of contributing across the organization. Thus, having a similar infra (e.g., GitHub/GitLab/etc) may help in this process.

Thus, as an overall goal here: the purpose of this section was to provide a way to assess the development process (mainly taking into account the first areas you mention such as CI, repositories, code review), but having under control the myriad of options out there and defining a common process within the organization.

At the InnerSource Commons we saw some cases where organizations are facing the issue of having isolated technological islands (e.g., certain programming language, certain tooling, certain process, etc). But when this is so specific to the team idiosyncrasy, and this team does not exist any more, then we have the maintenance issues, etc.

So, the proposal here is more focused on having a mature development process, but at the same time creating an homogeneous way of doing this across the organization.

What do you think @MaineC ?

Reply:
After thinking back and forth and comparing to open source projects, I agree with the point you are making, but would phrase it differently. .oO(this will take some typing to expand)

Regarding defining your own processes: I still think there is value in that. Think "how many trusted committer approvals do we require before merging?", "how should you approach us when making changes - do we always require an issue up front, or are we ok with receiving a pull request right away?", "exactly which issues will you run into when running our test suite?"

I believe giving projects some flexibility here helps reduce up-front coordination on how teams want to operate. (Think about the never ending discussion of where the curly braces should be put - over time, I have stopped worrying about that and simply follow the style that is already being used in the project.)

On the other end of the spectrum there are projects that move to exotic version control systems and build their own build systems. This happens in the open source world as well - with different degrees of success: Chrome (IIRC) at some point had to move to a more standard build tool to gain more contributors. The Linux kernel managed to establish their own version control system as the new normal.

In mature Inner Source projects me personally, I would expect trusted committers to be able to make a call which technology is best for the project - balanced with which technology is best to increase the likelihood of finding new contributors (use standard tooling, if you want to boost adoption and contributions).

In less mature organisations I would expect either everything being prescribed - or decisions on which to use being based not with the goal of increasing contribution levels, but based on team internal metrics only.

Right now, I have no idea how to transform this line of thinking into clear maturity levels - not even sure if that would help those adopting InnerSource who do not have years of Open Source experience - including years of searching for new contributors. It takes a while to realize that it takes people to write software.